File: solutions/go/space-age/1/space_age.go
// Package space provides functionality for converting ages between planets.
package space

// Planet represents a planet name.
type Planet string

const seconds = 31557600

var table = map[Planet]float64{
	"Earth":   seconds,
	"Mercury": 0.2408467 * seconds,
	"Venus":   0.61519726 * seconds,
	"Mars":    1.8808158 * seconds,
	"Jupiter": 11.862615 * seconds,
	"Saturn":  29.447498 * seconds,
	"Uranus":  84.016846 * seconds,
	"Neptune": 164.79132 * seconds,
}

// Age converts the earth's given age to the given planet's.
func Age(age float64, planet Planet) float64 {
	return age / table[planet]
}

File: solutions/clojure/space-age/1/src/space_age.clj
(ns space-age)

(def ^:private factors {:earth 1.0
                        :mercury 0.2408467
                        :venus 0.61519726
                        :mars 1.8808158
                        :jupiter 11.862615
                        :saturn 29.447498
                        :uranus 84.016846
                        :neptune 164.79132})
(def ^:private earth-year-seconds (* 60 60 24 365.25))
(defn- age-on [planet seconds] (/ seconds earth-year-seconds (factors planet)))

(defn on-earth [seconds] (age-on :earth seconds))
(defn on-mercury [seconds] (age-on :mercury seconds))
(defn on-venus [seconds] (age-on :venus seconds))
(defn on-mars [seconds] (age-on :mars seconds))
(defn on-jupiter [seconds] (age-on :jupiter seconds))
(defn on-saturn [seconds] (age-on :saturn seconds))
(defn on-uranus [seconds] (age-on :uranus seconds))
(defn on-neptune [seconds] (age-on :neptune seconds))
